contents/go/Go.md

```diff
@@ -0,0 +1,143 @@
+
+If you're unconvinced about Go, [the Go playground](https://play.golang.org/) can be used as a sandbox in which to write, build and execute code without installing Go on your machine.
+
+## Using Go
+As it builds on the foundations set by many popular and widely-used languages such as C, C++, Java and Python, much of Go's syntax draws from existing implementations and will be familiar to programmers looking to learn an additional language. However, Go also diverges explicitly from these other languages; listed below are some aspects of Go that may be unfamiliar to learners.
+
+### Arrays and slices
+Arrays are not often seen in Go code. This is because the size of a Go array must be declared at its creation, which limits how flexible an array can be.
+
+To create dynamically-sized arrays, Go introduces the concept of a slice. Slices are not arrays; rather, they are data structures that describe a piece of a separately-stored array. Slices are the topic of several Go blog entries; you can read more about [the internal layout of slices](https://blog.golang.org/go-slices-usage-and-internals) and [the mechanics of using slices](https://blog.golang.org/slices). Exercises on how to use slices are available in [A Tour of Go](https://tour.golang.org/moretypes/7).
+
+### `defer`, `panic` and `recover`
+In addition to traditional control flow mechanisms such as `if`, `for` and `switch`, Go also provides three other ways to alter the flow of a program: `defer`, `panic`, `recover`. These mechanisms can be conceived in terms of exception handling in Java and C++, but also have more general purpose uses.
+
+- `defer` pushes a function call to a list, and executes all functions on the list in last-in-first-out order after the surrounding function returns. It is frequently used for clean-up actions, such as to close files.
+- `panic` stops the ordinary flow of execution, executes any deferred functions, and returns to its caller as a call to `panic`. The `panic` continues up the stack until all functions in the goroutine have returned, after which the program crashes.
+- `recover` regains control of a panicking goroutine. When used inside a deferred function, a call to recover captures the value returned by `panic` and resumes normal execution.
+
+More information on these three mechanisms and their uses can be found on the [Go blog](https://blog.golang.org/defer-panic-and-recover).
+
+### Interfaces
```

Comment:
Since we're touching the OOP parts of Go, you can also add a section on how encapsulation works. https://code.tutsplus.com/tutorials/lets-go-object-oriented-programming-in-golang--cms-26540

Reply:
Hm, I'm not sure I want to explicitly mention this, as the section isn't about OOP in general. There's quite a lot to be said about Go's approach to OOP, but I don't think that I want to cover that here as it's somewhat prickly. Moreover, I intended for this section to walk through potential painpoints for a new learner of Go to look out for, and I think Go's encapsulation method is pretty straightforward. Will add it to the list of links though!